tests/test_commander_core.py

```diff
@@ -30,7 +30,7 @@ def __init__(self):
         self.sent = list()
 
         self._last_write = bytes()
-        self._modes = {}
+        self._modes = [None]
```

Comment:
To me it _seems_ that the issue is that 

```py
# In MockCommanderCoreDevice.write():
if self._modes[channel] is None:
```

should have always been

```py
if self._modes.get(channel) is None:
```

What do you think, @ParkerMc?

---

Either way, initializing a dict with a list doesn't feel right.

Reply:
@jonasmalacofilho Thanks for your feedback you're right it seems to be a better solution to leave the dict empty. I tested with your suggestion to get the mode and it works.